WordPress/Classes/Utility/DisplayableImageHelper.m

```diff
@@ -0,0 +1,184 @@
+#import "DisplayableImageHelper.h"
+
+static const NSInteger FeaturedImageMinimumWidth = 640;
+
+@implementation DisplayableImageHelper
+
+/**
+ Get the url path of the image to display for a post.
+
+ @param dict A dictionary representing a posts attachments from the REST API.
+ @return The url path for the featured image or nil
+ */
++ (NSString *)searchPostAttachmentsForImageToDisplay:(NSDictionary *)attachmentsDict
+{
+    NSArray *attachments = [[attachmentsDict dictionaryForKey:@"attachments"] allValues];
+    if ([attachments count] == 0) {
+        return nil;
+    }
+
+    NSString *imageToDisplay;
+
+    attachments = [self sanitizeAttachmentsArray:attachments];
+    NSSortDescriptor *descriptor = [[NSSortDescriptor alloc] initWithKey:@"width" ascending:NO];
+    attachments = [attachments sortedArrayUsingDescriptors:@[descriptor]];
+    NSDictionary *attachment = [attachments firstObject];
+    NSString *mimeType = [attachment stringForKey:@"mime_type"];
+    NSInteger width = [[attachment numberForKey:@"width"] integerValue];
+    if ([mimeType rangeOfString:@"image"].location != NSNotFound && width >= FeaturedImageMinimumWidth) {
+        imageToDisplay = [attachment stringForKey:@"URL"];
+    }
+
+    return imageToDisplay;
+}
+
+/**
+ Loops over the passed attachments array. For each attachment dictionary
+ the value of the `width` key is ensured to be an NSNumber. If the value
+ was an empty string the NSNumber zero is substituted.
+ */
++ (NSArray *)sanitizeAttachmentsArray:(NSArray *)attachments
```

Comment:
I'm not sure about this, but it seems a bit wasteful. Imagine a post with photo gallery from my summer vacation where I uploaded 200 pictures. The first item in the array will be a valid image, but this would still iterate through the entire array creating 200 dictionaries